crowbar_framework/app/models/api/storage.rb

```diff
@@ -15,6 +15,45 @@
 #
 
 module Api
-  class Storages
+  class Storage < Tableless
+    def repocheck
+      {}.tap do |ret|
+        ret["ceph"] = {
+          "available" => true
+        }
+
+        platform = Api::Upgrade.new.target_platform(platform_exception: :ceph)
+
+        ["os", "ceph"].each do |feature|
+          ceph_node_architectures.each do |architecture|
+            unless ::Crowbar::Repository.provided_and_enabled?(feature,
+                                                               platform,
+                                                               architecture)
+              ::Openstack::Upgrade.enable_repos_for_feature(feature, Rails.logger)
+            end
+            available, repolist = ::Crowbar::Repository.provided_and_enabled_with_repolist(
+              feature, platform, architecture
+            )
+            ret["ceph"]["available"] &&= available
+            ret["ceph"]["repos"] ||= {}
+            ret["ceph"]["repos"].deep_merge!(repolist.deep_stringify_keys)
+          end
+        end
+      end
+    end
+
+    protected
+
+    def ceph_node_architectures
+      ceph_proposal = Proposal.find_by(barclamp: "ceph")
+      return [] if ceph_proposal.nil?
+      return [] unless ceph_proposal.status == "ready"
```

Comment:
(btw, interesting trick to find the applied proposals; I would have gone a different approach, by looking at RoleObject, but this may be better)